### DISCUSSION TITLE
Internet Connection

### DISCUSSION THREAD

**Discussion by 26zl** [HotCakeX/Harden-Windows-Security#362] [Q&A]
Hi! After running the windows defender, firewall and networking categories of the script, i no longer have any internet connection. It does connect to the network but i do not get any internet both with wifi and cable. When running unprotect-windows security command i get Invoke-restMethod: No such host is known. I have a restore point but that restore point is a month old i could use it but want to try any other possible solution first. (Have used it before without problems with the same pc).

**Comment by HotCakeX** [comment]
Hi, you can run the module completely offline too of course as mentioned [in the docs](https://github.com/HotCakeX/Harden-Windows-Security/wiki/Harden%E2%80%90Windows%E2%80%90Security%E2%80%90Module).
You need to download LGPO, Microsoft security baselines and Microsoft 365 apps security baselines zip files from here:
https://www.microsoft.com/en-us/download/details.aspx?id=55319&msockid=20658dfd37856ec720e498f1367a6f64

And then use the `Protect-WindowsSecurity -GUI` to start the GUI, navigate to the offline section, browse for the 3 zip files you downloaded and then use the Unprotect tab to remove the protections. No Internet connection required.

Also every policy and measure used is mentioned in the Readme, none of them disable Internet connection, I'm fairly certain there must be another factor that caused that on your system.

**Reply by HotCakeX** [reply]
Here are the direct links to the 3 files

* https://download.microsoft.com/download/8/5/C/85C25433-A1B0-4FFA-9429-7E023E7DA8D8/Windows%2011%20v24H2%20Security%20Baseline.zip

* https://download.microsoft.com/download/8/5/C/85C25433-A1B0-4FFA-9429-7E023E7DA8D8/Microsoft%20365%20Apps%20for%20Enterprise%202306.zip

* https://download.microsoft.com/download/8/5/C/85C25433-A1B0-4FFA-9429-7E023E7DA8D8/LGPO.zip

**Reply by HotCakeX** [reply]
I have one idea, is your Firewall profile set to domain? That's normally not the case because only computers part of the domain of a Windows Servers might use it. The Harden Windows Security module wouldn't make sense to be used in a domain-controlled environment since the domain controller must deploy the policies and not the clients.

The Firewall category disables inbound/outbound connections for the Domain profile. Public profile offers the highest security, the Windows Networking category sets all network connections profiles to public.

<br>

![image](https://github.com/user-attachments/assets/066e0b2b-3ec9-46f5-b3d2-588ae3a3141b)

<br>

<br>

You can check your network profile from the Microsoft Defender app

![image](https://github.com/user-attachments/assets/386cc6b5-d4a8-4055-a424-66a3e9886690)

**Reply by 26zl** [reply]
There was nothing wrong with your program; this was an issue on my end and a user configuration error. All good. Sorry for the hassle, and I’m glad you replied. I always learn something from you, haha. Thanks again!

**Comment by 26zl** [comment]
Thank you very much i managed to fix it with your solution, thanks a lot. But for some weird reason system confirm compliance does not work for me know. This the error message i get. 
![Skjermbilde 2024-10-14 001844](https://github.com/user-attachments/assets/2754e88c-ad14-4486-84de-638b69a975bb)

**Reply by HotCakeX** [reply]
Anytime, happy to help ^^

Based on this discussion, i'm going to move the section that sets all network connection profiles to public, from the Windows Networking category to the Windows Firewall category, it will prevent this situation from happening in the future.

About that error you mentioned, i found the root cause of it. It happens when a service called "Workstation" is disabled. It is not disabled by default or by the Harden Windows Security Module, only user configurations would have disabled it.

Here in this screenshot i could easily reproduce it. This is another thing I'm going to change in the next version. The compliance check and anything else that might not work when that default service is disabled, will check its status and enable/start it again if necessary.

For now you can start it and set its startup status to automatic and then run the command again.

![image](https://github.com/user-attachments/assets/82b7dc7b-133d-49fc-823b-67e72bae441f)

**Reply by 26zl** [reply]
Thank you so much! Yeah, Workstation was disabled. I was trying out some things with SMB and didn’t realize that Workstation was turned off. It works know.

**Reply by HotCakeX** [reply]
Yw! You won't need to remmeber to have that service enabled. In the new update i just released it happens automatically for you

https://github.com/HotCakeX/Harden-Windows-Security/releases/tag/Hardening-Module-v.0.6.7

**Reply by 26zl** [reply]
That was quick! 😂 Well done!

**Reply by HotCakeX** [reply]
Thanks 🙏
